tiled/access_control/access_policies.py

```diff
@@ -412,3 +415,178 @@ async def filters(
 
         queries.append(query_filter(identifier, tag_list))
         return queries
+
+
+class Data(BaseModel):
+    token: str
+    audience: Optional[str]
+    access_blob: Optional[AccessBlob]
+
+
+class Input(BaseModel):
+    input: Data
+
+
+class Decision(BaseModel):
+    result: Union[List[str], bool]
+
+
+class ExternalPolicyDecisionPoint(AccessPolicy):
+    def __init__(
+        self,
+        authorization_provider: HttpUrl,
+        node_access: str,
+        filter_nodes: str,
+        scopes_access: str,
+        audience: str,
+        provider: Optional[str] = None,
+    ):
+        self.authorization_provider = authorization_provider
+
+        self._node_access = urljoin(
+            str(self.authorization_provider),
+            urlsplit(node_access).path,
+        )
+        self._filter_nodes = urljoin(
+            str(self.authorization_provider),
+            urlsplit(filter_nodes).path,
+        )
+        self._scopes_access = urljoin(
+            str(self.authorization_provider),
+            urlsplit(scopes_access).path,
+        )
+
+        self.audience = audience
+        self.provider = provider
+
+    def _identifier(self, principal) -> str:
+        if principal.type == PrincipalType.service:
```

Comment:
(not knowing enough about AccessPolicy)- should we be returning anything if it's not an external Principal? As an authorization provider, we can't verify that the principal is authenticated if it's not a token